cmd/apprepository-controller/server/controller.go

```diff
@@ -580,13 +586,49 @@ func ttlLifetimeJobs(config Config) *int32 {
 }
 
 // cronJobName returns a unique name for the CronJob managed by an AppRepository
-func cronJobName(namespace, name string) string {
-	return fmt.Sprintf("apprepo-%s-sync-%s", namespace, name)
+func cronJobName(namespace, name string, addDash bool) string {
+	// the "apprepo--sync-" string has 14 chars, which leaves us 52-14=38 chars for the final name
+	maxNamesapceLength, rem := (MAX_CRONJOB_CHARS-14)/2, (MAX_CRONJOB_CHARS-14)%2
+	maxNameLength := maxNamesapceLength
+	if rem > 0 && !addDash {
+		maxNameLength++
+	}
+	truncatedName := fmt.Sprintf("apprepo-%s-sync-%s", truncateAndHashString(namespace, maxNamesapceLength), truncateAndHashString(name, maxNameLength))
+	if addDash {
+		truncatedName = fmt.Sprintf("%s-", truncatedName)
+	}
+	return truncatedName
 }
 
 // deleteJobName returns a unique name for the Job to cleanup AppRepository
 func deleteJobName(namespace, name string) string {
-	return fmt.Sprintf("apprepo-%s-cleanup-%s", namespace, name)
+	// the "apprepo--cleanup--" string has 18 chars, which leaves us 52-18=34 chars for the final name
+	maxNamesapceLength, rem := (MAX_CRONJOB_CHARS-18)/2, (MAX_CRONJOB_CHARS-18)%2
+	maxNameLength := maxNamesapceLength
+	if rem > 0 {
+		maxNameLength++
+	}
+
+	truncatedName := fmt.Sprintf("apprepo-%s-cleanup-%s-", truncateAndHashString(namespace, maxNamesapceLength), truncateAndHashString(name, maxNameLength))
+	return truncatedName
+}
```

Comment:
Looks like these two functions could easily be a single one? `cronJobName(namepsace, name, action string, addDash bool)` ? (where `action` is passed in as "cleanup" or "sync"). You could still have the separate `deleteJobName` which would just call with the correct action? Not sure, maybe not worth it.